### ISSUE TITLE
Incorrect register class of instruction pointer register

### ISSUE THREAD

**Issue by rgov** [zyantific/zydis#33]
```
$ ZydisInfo -64 488b05b6310500
...
== [ OPERANDS ] ============================================================================================
##       TYPE  VISIBILITY  ACTION      ENCODING   SIZE  NELEM  ELEMSZ  ELEMTYPE                        VALUE
--  ---------  ----------  ------  ------------   ----  -----  ------  --------  ---------------------------
 0   REGISTER    EXPLICIT       W     MODRM_REG     64      1      64       INT                          rax
 1     MEMORY    EXPLICIT       R      MODRM_RM     64      1      64       INT  TYPE  =                 MEM
                                                                                 SEG   =                  ds
                                                                                 BASE  =                 rip
                                                                                 INDEX =                none
                                                                                 SCALE =                   0
                                                                                 DISP  =  0x00000000000531B6
```

Here this instruction's memory operand uses `%rip` as its base register. If I modify ZydisInfo to print the register class, as in `ZydisRegisterGetClass(instruction->operands[i].mem.base)`, I get `ZYDIS_REGCLASS_INVALID`, which seems incorrect. I expect `ZYDIS_REGCLASS_IP`.

**Comment by rgov** [comment]
More simply: `ZydisRegisterGetClass(ZYDIS_REGISTER_EIP) = ZYDIS_REGCLASS_INVALID`.

**Comment by rgov** [comment]
In `enum ZydisRegisters` (Register.h), we have the order of instruction pointer references given, in ascending order, `ZYDIS_REGISTER_IP,     ZYDIS_REGISTER_EIP,    ZYDIS_REGISTER_RIP`.

But the corresponding entry in `registerMap` (Register.c) assumes `RIP` is "lo" and `IP` is "high":

```
    { ZYDIS_REGCLASS_IP       , ZYDIS_REGISTER_RIP    , ZYDIS_REGISTER_IP     ,   0   ,   0 },
```

This makes `ZydisRegisterGetClass` fail to enumerate the range correctly. I'll submit a pull request.

**Comment by rgov** [comment]
Fixed in 8c5e216e11fd478bc027dc33bffcaae788a0903f.

**Comment by athre0z** [comment]
Nevertheless, thanks for taking the time to look into this yourself. We appreciate it!
